### PR TITLE
[fix] oscar theme, remove inline style attributes (CSP compliants) 

### PR DESCRIPTION
**Pull request by return42**
Inline styles are blocked by default with Content Security Policy (CSP).  Move the rest of inline styles to CSS and correct the HTML template of the oscar preference page.  Fix of the error messages from Chrome based browsers::

    .. Refused to apply inline style because ..

see comment: https://github.com/asciimoo/searx/issues/1949#issuecomment-628537242

[Merged](https://github.com/return42/searx/projects/1) into the [searx-next branch](https://github.com/return42/searx/tree/searx-next) /  preview: https://darmarit.org/searx/preferences

### CODE REVIEW

**Comment by return42** [comment]
@asciimoo can you please merge this PR first before adding any more patches to the oscar theme.  I am tired to wake up every morning and to have solve conflicts in the min.js files first / thanks!

**Comment by return42** [comment]
BTW 2c6531b2330a5c breaks the pep8 test of all the PRs awaiting in the pipe. See https://github.com/asciimoo/searx/commit/2c6531b2330a5c8ca986379dbad8812a6c6d3e49#diff-65ed3496c99aad64ac6a6e71504e0101R40

OT / it is not only this PR, and I am really sorry to say:

I'm getting more and more frustrated. I invested months of work to improve searx substantially ([see searx-next merged PRs](https://github.com/return42/searx/projects/1)). But my PRs are ignored, which entails considerable maintenance costs: Instead of reviewing PRs, changes are shot in, which additionally increase the organizational overhead. Something is going wrong if we are three developers here and generate an organizational effort that is otherwise only known from sluggish, large companies.

**Comment by HLFH** [comment]
Hi @asciimoo,

> I'm getting more and more frustrated. I invested months of work to improve searx substantially (see searx-next merged PRs). But my PRs are ignored, which entails considerable maintenance costs: Instead of reviewing PRs, changes are shot in, which additionally increase the organizational overhead. Something is going wrong if we are three developers here and generate an organizational effort that is otherwise only known from sluggish, large companies.

Can this be fixed @asciimoo?

Thanks!

**Comment by return42** [comment]
WTH I am saying!?! .. commit 4ca0d8cb0fd is next one, directly shot in without a review process.

I feel like i am trapped in the groundhog-day time loop .. next manual merge cc721b5d of a conflict caused by 4ca0d8cb0fd.

@HLFH thanks for your support .. but I guess all we are writing here goes directly into @asciimoo 's spam folder .. also tried personal mails, seems also going into spam .. sometimes (really rare) I get a answer (with little informative value, not really a knowledge exchange) .. mostly I am ignored.

**Comment by return42** [comment]
OK, another conflict and this PR is ignored .. I gave up.

File: searx/static/themes/oscar/css/logicodev.min.css
Comment (SuperSandro2000) [inline searx/static/themes/oscar/css/logicodev.min.css]: If we wouldn't commit the minified files this PR wouldn't conflict as often but it also could have been merged a while ago.

Comment (return42) [reply searx/static/themes/oscar/css/logicodev.min.css]: Yes minified and binaries are causing problems in collaboration .. as far as I know @asciimoo do not want to have a build process / sorry, can't remember where I have read this.  May be he as changed his opinion about .. @asciimoo ?

Ah, forgot, we will not get an answer ..

**Comment by asciimoo** [comment]
> @asciimoo can you please merge this PR first before adding any more patches to the oscar theme. I am tired to wake up every morning and to have solve conflicts in the min.js files first / thanks!

My main priority was to resolve issues required to release 1.0.0 , sorry if it caused problems to you. Probably the conflicts are mostly in the minified resources what can be easily resolved.

> I'm getting more and more frustrated.

Im so sad reading this. Apologies for this. If it's my fault, please help me with suggesting solutions to avoid such situations.

> I invested months of work to improve searx substantially (see searx-next merged PRs). But my PRs are ignored

They are not ignored. I'm trying to do everything in my limited time and sometimes it's really hard to prioritize and allocate enough time to digest multiple thousand lines long PRs and 30-40-50 mails daily basis just about searx while I also have other projects/jobs.

> also tried personal mails, seems also going into spam 

I think It isn't my fault that gmail classified your emails as spam. Probably It's a mail server misconfiguration issue.

> @asciimoo do not want to have a build process

I don't think I've ever said something like this, but whether it happened or not, now I'd like to explicitly state that I'm not against any kind of improvement that can help the development or the job of the contributors/maintainers.
I know that there are so many unresolved issues around the development, let's focus on finding solutions for these.

**Comment by return42** [comment]
@asciimoo thanks for taking this into account.  I will answer the more general aspects in https://github.com/asciimoo/searx/issues/2025

> My main priority was to resolve issues required to release 1.0.0 

Can we please add CSP to that list.  I have seen that others add exceptions for the `style` to the HTTP headers, but I think it is better to isolate style into CSS files.

> I think It isn't my fault that gmail classified your emails as spam. Probably It's a mail server misconfiguration issue.

Its first time I hear about such problems with my mail server and I do not know how to trainee g-mails's spam filter.  I know from many filters, that they wont filter out mails from senders which are in the _contatcs_. May it helps if you add my e-mail address to your contacts.

> I don't think I've ever said something like this ..I know that there are so many unresolved issues around the development, let's focus on finding solutions for these.

Ah, OK .. that is fine for me, lets add a build process to eliminate conflicts in build products.  I started issue https://github.com/asciimoo/searx/issues/2034 .. if you have some ideas, your remarks are welcome / or better say: your remarks are needed :)

**Comment by dalf** [comment]
Except the conflicting files, LGTM.

**Comment by asciimoo** [comment]
`<style>` tags in the HTML don't compromise the CSP?

**Comment by dalf** [comment]
> ```<style>``` tags in the HTML don't compromise the CSP?

@asciimoo , are you referencing this pattern 
```
    <noscript>
        <style type="text/css">
...
        </style>
    </noscript>
```
?

True, no javascript doesn't mean the browser doesn't support CSP (typically with Firefox with the [uMatrix](https://addons.mozilla.org/en/firefox/addon/umatrix/) extension).
* One solution is to add a file ( .less then produce a .css).
* Another solution is to add a ```js``` CSS class like in the simple theme: https://github.com/asciimoo/searx/blob/6fbba63c83b18a652e571ce5a803773cd72e5b6a/searx/static/themes/simple/js/searx.head.js#L39
* A third solution:  add a [```nonce-<base64-value>```](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src) value in the CSP header, which is related to https://github.com/asciimoo/searx/issues/715

For now, this PR fixes the more common use case: a browser with Javascript and CSP support.

**Comment by asciimoo** [comment]
> @asciimoo , are you referencing this pattern

Yup

> For now, this PR fixes the more common use case: a browser with Javascript and CSP support.

Okay, I have no problem with merging the current state, but keep in mind these `<style>` tags too.

**Comment by SuperSandro2000** [comment]
You can find a very good documentation [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src). The paragraph ``Unsafe inline styles`` is probably the most intersting.

**Comment by return42** [comment]
I just wanted to remark, that I think; setting `style-src 'unsafe-inline'` is not really the best solution.

**Comment by SuperSandro2000** [comment]
I did not want to suggest that. It is just Mozillas doc on the CSP ``style-src`` and what falls under ``'unsafe-inline'``.

**Comment by asciimoo** [comment]
@return42 could you please rebase it for the last time to be able to merge it?

**Comment by return42** [comment]
> could you please rebase it for the last time to be able to merge it?

@asciimoo yes, done / see f14a7add

**Comment by asciimoo** [comment]
Great, thanks!
Could you please rebase in the future instead of merging back the master into the feature branch? It keeps the git tree clean. Currently our master branch looks like a busy train station: 
x

**Review by asciimoo — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
The image looks like art. 😂

**Comment by return42** [comment]
> Could you please rebase in the future instead of merging back the master into the feature branch? 

I see, sorry .. ATM this is a result of the way I [merge PRs](https://github.com/return42/searx/projects/1) with [searx-next branch](https://github.com/return42/searx/tree/searx-next).

I do not have a clue how I could rebase a public branch (PR) which is already merged into another public branch (searx-next)    

@SuperSandro2000, @dalf and me discussed this subject beginning near by https://github.com/asciimoo/searx/pull/1985#issuecomment-649460229

Does anyone have a clue, how I can organize my searx-next branch and the branches from PRs without getting in conflict with the [golden rule; “No one shall rebase a shared branch”](https://www.daolf.com/posts/git-series-part-2)? I guess I'm not the only one asking this question.

**Comment by asciimoo** [comment]
What is the purpose of your searx-next branch? Cherry-picking is also an option which can work here I think.

**Comment by return42** [comment]
> What is the purpose of your searx-next branch?

When I had a few [PRs in the loop](https://github.com/return42/searx/projects/1), I started that branch.  It purpose is to merge PRs aimed to merge at asciimoo/master (in general, not restricted to PRs of mine).

This branch has it's reference instance at https://darmarit.org/searx/ and the documentation of the _brand_ is available at https://return42.github.io/searx/

> Cherry-picking is also an option which can work here I think.

This is was others also suggested, but sooner or later it will end in a mess, so I prefer to track branches (pull requests).

**Comment by asciimoo** [comment]
> s branch has it's reference instance at https://darmarit.org/searx/ 

I see, thanks for the clarification.

> but sooner or later it will end in a mess

Maintaining two different forks always has its challenges, but the current solution also causes mess in our side. Please try cherry-picking and lets continue the brainstorming if it's too uncomfortable.

**Comment by return42** [comment]
> but the current solution also causes mess in our side. ... Please try cherry-picking 

I will stop adding PRs to searx-next .. as long as I do not have a more comfortable method than cherry-pickng / I have to think about it .. remarks & hints are welcome ..

**Comment by asciimoo** [comment]
What is exactly the problem with cherry-pick? I think it's quite convenient to create a clear feature branch, pick some commits and create/update the PR, you can even rebase that branch without conlficting with your searx-next branch.

**Comment by return42** [comment]
@asciimoo thanks for taking up the brainstorm with me .. I have a doubt that a understand your suggestions. Some word to clarify ..

On the one side; I understand that you like to see a branch cleared up before merge into master.  In my self-next branch there are two PRs open which not have been merged into asciimoo/master (#1985 and #1938).  [I can clean up theses branches if I got the OK for a merge](https://github.com/asciimoo/searx/pull/2038#issuecomment-653401062).  Since there are not many clones from searx--next exists, it should also OK to rebase searx-next in context of this both PRs .. I think this problem is solved.

On the other side, I want to progress with brainstorming about maintaining a branch like searx-next.

The motivation of the searx-next branch is to have a branch with a reference installation and this branch should also a testing area for PRs (other branches) before I (we) merge them into asciimoo/master . For smaller PRs without any special in, we do not need such a _reference installation_ but there are also examples where it is good to have one, e.g. when I try to fix CSP errors or just test some issue users added to our issue tracker (often, they use an unmaintained instance and the reported bug have already been fixed).  BTW: is there a chance to have [searx.me](https://github.com/asciimoo/searx/blob/master/Makefile#L7) back?

> What is exactly the problem with cherry-pick? 

It does not track .. if you have a bunch of PRs (branches) which are patched in the review phase, you have to track all branches manual.

> I think it's quite convenient to create a clear feature branch, pick some commits and create/update the PR, you can even rebase that branch without conlficting with your searx-next branch.

I haven't understand in deep .. what I know (may I'm wrong) .. when you track a branch, and this branch is rebased or the history is changed in any manner, you lost the anchor where the external branch was started initial and than you do not have a chance to merge (pull) from that branch .. I'm wrong .. or did I misunderstood your suggestion?

**Pull request merged**
